src/audio/copier/copier.c

```diff
@@ -723,24 +708,17 @@ static void copier_free(struct comp_dev *dev)
 {
 	struct copier_data *cd = comp_get_drvdata(dev);
 	int i;
-	bool flag = true;
+
+	if (dev->ipc_config.type == SOF_COMP_HOST && !cd->ipc_gtw) {
+		host_zephyr_free(cd->hd);
+		rfree(cd->hd);
+	}
```

Comment:
is it possible that `dev->ipc_config.type == SOF_COMP_HOST && cd->ipc_gtw`? Does anything have to be freed then?

Reply:
please check latest, I added with:
if (dev->ipc_config.type != SOF_COMP_HOST || cd->ipc_gtw)